flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2628,14 +2689,20 @@ class ScalarExprLowering {
     return gen(x);
   }
 
+  llvm::SmallVector<mlir::Attribute> &getRawVals() { return rawVals; }
+  mlir::Type getRawType() { return rawType; }
```

Comment:
Is it possible to pass these as arguments to `createDenseGlobal()`? We don't carry around much special state in an SEL object.

Reply:
I'm not sure what you're suggesting. Just pass in a reference for these in the SEL constructor? I think we need to build this up in SEL during the expression traversal.

Reply:
Generally, it's not a good idea to cache transient state in the lowering object. That is because expressions are recursive, so the state in one context might be corrupted as the tree is walked. Having `rawType` and `rawVals` dangle off a `this` pointer can break under recursion.

A more robust approach is to build these things on the call stack and pass them around as arguments to where they are used.